relay-profiling/src/android/chunk.rs

```diff
@@ -0,0 +1,169 @@
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    debug_meta: Option<DebugMeta>,
+
+    client_sdk: ClientSdk,
+
+    #[serde(default)]
+    duration_ns: u64,
+    #[serde(default)]
+    timestamp: f64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Chunk {
+    #[serde(flatten)]
+    metadata: Metadata,
+
+    #[serde(default, skip_serializing)]
+    sampled_profile: String,
+
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    js_profile: Option<ProfileData>,
+
+    #[serde(default = "Chunk::default")]
+    profile: AndroidTraceLog,
+
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    measurements: Option<HashMap<String, Measurement>>,
+}
+
+impl Chunk {
+    fn default() -> AndroidTraceLog {
+        AndroidTraceLog {
+            data_file_overflow: Default::default(),
+            clock: Clock::Global,
+            elapsed_time: Default::default(),
+            total_method_calls: Default::default(),
+            clock_call_overhead: Default::default(),
+            vm: Vm::Dalvik,
+            start_time: Utc::now(),
+            pid: Default::default(),
+            gc_trace: Default::default(),
+            threads: Default::default(),
+            methods: Default::default(),
+            events: Default::default(),
+        }
+    }
+
+    fn parse(&mut self) -> Result<(), ProfileError> {
+        let profile_bytes = match BASE64_NOPAD.decode(self.sampled_profile.as_bytes()) {
+            Ok(profile) => profile,
+            Err(_) => return Err(ProfileError::InvalidBase64Value),
+        };
+        self.profile = match android_trace_log::parse(&profile_bytes) {
+            Ok(profile) => profile,
+            Err(_) => return Err(ProfileError::InvalidSampledProfile),
+        };
+        Ok(())
+    }
+}
+
+fn parse_chunk(payload: &[u8]) -> Result<Chunk, ProfileError> {
+    let d = &mut serde_json::Deserializer::from_slice(payload);
+    let mut profile: Chunk = match serde_path_to_error::deserialize(d) {
+        Ok(p) => p,
+        Err(e) => {
+            println!("{:#?}", e);
+            return Err(ProfileError::InvalidJson(e));
+        }
+    };
+    // .map_err(ProfileError::InvalidJson)?;
+
+    if let Some(ref mut js_profile) = profile.js_profile {
+        js_profile.normalize(profile.metadata.platform.as_str())?;
+    }
+
+    if !profile.sampled_profile.is_empty() {
+        profile.parse()?;
+    }
+
+    if profile.profile.events.is_empty() {
+        return Err(ProfileError::NotEnoughSamples);
+    }
+
+    if profile.profile.elapsed_time > MAX_PROFILE_DURATION {
+        return Err(ProfileError::DurationIsTooLong);
+    }
+
+    // Use duration given by the profiler and not reported by the SDK.
+    profile.metadata.duration_ns = profile.profile.elapsed_time.as_nanos() as u64;
+    profile.metadata.timestamp = profile
+        .profile
+        .start_time
+        .timestamp_nanos_opt()
+        .unwrap_or_default() as f64
+        * 1e-9;
```

Comment:
nit: you could avoid the manual conversion.

let start_time_duration = Duration::from_nanos(
    profile.profile.start_time.timestamp_nanos_opt().unwrap_or_default() as u64,
);
profile.metadata.timestamp = start_time_duration.as_secs_f64();